### PR TITLE
[feature-wip](new-scan) support Json reader

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #12574
This pr adds `NewJsonReader` which implements GenericReader interface to support read json format file.

TODO:
1. modify `_scann_eof` later.
2. Rename `NewJsonReader` to `JsonReader` when `JsonReader` is deleted.

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.92 seconds
 load time: 570 seconds
 storage size: 17154644820 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221021195440_clickbench_pr_32519.html

File: be/src/vec/exec/format/json/new_json_reader.cpp [hunk 1]
```diff
@@ -0,0 +1,753 @@
...
+    _bytes_read_counter = ADD_COUNTER(_profile, "BytesRead", TUnit::BYTES);
+    _read_timer = ADD_TIMER(_profile, "ReadTime");
+    _file_read_timer = ADD_TIMER(_profile, "FileReadTime");
+}
+
+Status NewJsonReader::init_reader() {
+    RETURN_IF_ERROR(_get_range_params());
+
+    LOG(INFO) << "--ftw: strip_out_array" << _strip_outer_array;
+    LOG(INFO) << "--ftw: read_json_by_line" << _read_json_by_line;
```
[the remark this answers is not in the document]
Comment (BePPPower) [reply be/src/vec/exec/format/json/new_json_reader.cpp]: done

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.77 seconds
 load time: 574 seconds
 storage size: 17154644803 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221023175947_clickbench_pr_32848.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.85 seconds
 load time: 559 seconds
 storage size: 17154887228 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221026031737_clickbench_pr_34041.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**